config/src/network_id.rs

```diff
@@ -1,12 +1,62 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
+use crate::config::RoleType;
+use libra_types::PeerId;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
+/// A grouping of common information between all networking code for logging.
+/// This should greatly reduce the groupings between these given everywhere, and will allow
+/// for logging accordingly.  TODO: Figure out how to split these as structured logging
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct NetworkContext {
+    network_id: NetworkId,
+    role: RoleType,
+    peer_id: PeerId,
+}
+
+impl std::fmt::Display for NetworkContext {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
+            formatter,
+            "[{},{},{}]",
+            self.network_id.to_string(),
+            self.role.as_str(),
+            self.peer_id.short_str()
+        )
+    }
+}
+
+impl NetworkContext {
+    pub fn new(network_id: NetworkId, role: RoleType, peer_id: PeerId) -> NetworkContext {
+        NetworkContext {
+            network_id,
+            role,
+            peer_id,
+        }
+    }
+
+    pub fn network_id(&self) -> NetworkId {
+        self.network_id.clone()
+    }
+    pub fn peer_id(&self) -> PeerId {
+        self.peer_id
+    }
+    pub fn role(&self) -> RoleType {
```

Comment:
I would probably do

```rust
pub fn network_id(&self) -> &NetworkId { &self.network_id }
```

since it's (relatively) expensive to clone the heap allocated string